### ISSUE TITLE
[BUG] Unnecessary TimeSeries object updates

### ISSUE THREAD

**Issue by raoulcollenteur** [pastas/pastas#250]
**Describe the bug**
There's sometimes (a large number of) unnecessary TimeSeries object updates. This can for example be seen in the example model of this notebook:

https://pastas.readthedocs.io/en/dev/concepts/handling_timeseries.ipynb.html

It seems to have something to do with the tmin and tmax, maybe something with the offsets when extending time series. Possibly also some comparison in _update_settings. I think the notebook is a good example to work from.

**EDIT:** 
This is also a good example.

https://pastas.readthedocs.io/en/dev/examples/010_multiple_wells.ipynb.html

Maybe it has something to do with ml.plots.results....

**To Reproduce**
See above

**Expected behavior**
I think this should only be done once.

**Python package version**
Python version: 3.7.8 | packaged by conda-forge | (default, Jul 31 2020, 02:37:09)
[Clang 10.0.1 ]
Numpy version: 1.18.5
Scipy version: 1.4.0
Pandas version: 1.1.2
Pastas version: 0.16.0b
Matplotlib version: 3.1.3

**Comment by rubencalje** [comment]
Yes, I've noticed similar behaviour. Do not know if this was always the case, or just recently. It would be great if you can solve this.

**Comment by raoulcollenteur** [comment]
I looked into this this morning and found what causes it I think. It started after pulling PR #237. Throughout model.py time offsets are now added (e.g., sim_index) but not always (ml.get_contributions). The tmin and tmax stored in ml.settings are therefore not the tmin and tmax of the simulation anymore. This should not be the case I think.

The tmin and tmax stored in the ml.settings (used if a user provides nothing) are now updated after sim_index is created. Honestly I somehow feel all this is overcomplicated and could be done much easier. I've spend 6 hours on this now and it's almost impossible for me to figure out what happens with all these time series and get it right.

See the PR for something that seems to work..

**Comment by rubencalje** [comment]
> I looked into this this morning and found what causes it I think. It started after pulling PR #237.

Hmm, then it is probably my fault. I deleted some changes to tmin and tmax, to make the code simpler, and I reasoned then it was allowed. I probably deleted too much. Shall I look into it?

**Comment by rubencalje** [comment]
You are right that tmin is updated before performing simulations. This must be done to incorporate the warmup-period. But this should be done in the same manner in ml.simulate() and ml.get_contributions(). Therefore I made some changes in my last commit. This should solve the problem.
I also found out that tmax was handled differently in ml.simulate() and ml.get_contributions(). This had no influence on the result, but also caused multiple timeseries-updates.

It should work now.
